src/pr.test.ts

```diff
@@ -0,0 +1,42 @@
+import { ruleFactory } from './index';
+
+describe('array method object syntax', () => {
+  describe('map', () => {
+    it('execute', () => {
+      const rules = [
+        'idList = [12, 34, 56]',
+        {
+          map: 'idList',
+          run: [
+            {
+              if: 'GET(ITER.item, userScore) < 0',
```

Comment:
I don't know if I'm 💯% on the `ITER` symbol, 🤔

Reply:
I'm sure there's a better name. Using the input context seemed like the easiest way to give the parser access to the array element being mapped.